### PR TITLE
fix(embeds): fix embeds problem

### LINKED ISSUE #667: soundcloud embeds do not work
**Description**
Soundcloud embeds do not work with markdown. 

**To Reproduce**
Try to embed https://soundcloud.com/cheryl-lin-fielding/chanson-pour-jeanne into a gdoc or github markdown repo! The pipeline will miss the embed

test/fixtures/embed.md

**Expected behavior**
the abovementioned soundcloud link should be processed as an embed

- what kind of HTML do you want to see
```
<div class="embed  embed-has-url embed-has-title embed-has-image embed-has-description">
  <a href="https://soundcloud.com/cheryl-lin-fielding/chanson-pour-jeanne">
    <span class="title">Chabrier: Chanson pour Jeanne - Efrain Solis, Baritone & Cheryl Lin Fielding, Piano</span>
  </a>
  <img src="https://i1.sndcdn.com/avatars-000221968272-4i1fcm-t500x500.jpg" alt="Chabrier: Chanson pour Jeanne - Efrain Solis, Baritone & Cheryl Lin Fielding, Piano" class="image">
    <p class="description">Efrain Solis, Baritone
Cheryl Lin Fielding, Piano
Live performance</p>
<div>
```
**Actual behavior**
The pipeline fails to see this as an embed, so it's seen as: 

      -          "title": [null]
      -          "type": "link"
      -          "url": "https://soundcloud.com/cheryl-lin-fielding/chanson-pour-jeanne"
      +          "type": "paragraph"
               }
             ]

This is the output from index.test.js of the pipeline which tests embeds, type should be embed not paragraph.

### PR DESCRIPTION
**Pull request by MarquiseRosier**
fixed whitelisting issue; which was preventing embeds

fix #667

### CODE REVIEW

**Comment by MarquiseRosier** [comment]
The problem is with this line! 

https://github.com/adobe/helix-pipeline/blob/c78860e469f52cd8d88472fac9a371dec1efd5b0/src/html/find-embeds.js#L101

The whitelist looks like this: 
```['www.youtube.com, spark.adobe.com, unsplash.com, soundcloud.com']```

When we call ```whitelist.split(',')``` , it ends up like this. 

```[ 'www.youtube.com',
  ' spark.adobe.com',
  ' unsplash.com',
  ' soundcloud.com' ]```

and so ```"soundcloud.com"``` does not equal ```" soundcloud.com"```

File: src/html/find-embeds.js [hunk 1]
```diff
@@ -92,13 +92,13 @@ function internalImgEmbed({ type, children }, base, contentext, resourceext) {
 }
 
 function embed(uri, node, whitelist = '', datawhitelist = '', logger) {
-  if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, datawhitelist.split(','))) {
+  if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, datawhitelist.split(', '))) {
```
Comment (stefan-guggisberg) [inline src/html/find-embeds.js:95]:
```suggestion
  if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, datawhitelist.split(',').map((s) => s.trim()))) {
```

File: src/html/find-embeds.js [hunk 2]
```diff
@@ -92,13 +92,13 @@ function internalImgEmbed({ type, children }, base, contentext, resourceext) {
...
     const children = [{ ...node }];
     node.type = 'dataEmbed';
     node.children = children;
     node.url = URI.serialize(uri);
     delete node.value;
-  } else if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, whitelist.split(','))) {
+  } else if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, whitelist.split(', '))) {
```
Comment (stefan-guggisberg) [inline src/html/find-embeds.js:101]:
```suggestion
  } else if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, whitelist.split(',').map((s) => s.trim()))) {
```

**Comment by MarquiseRosier** [comment]
> Great catch! I made a small change to make the code more lenient regarding the format of the `whitelist` string.

Thank you @stefan-guggisberg and great idea too!

File: src/html/find-embeds.js [hunk 3]
```diff
@@ -92,13 +92,13 @@ function internalImgEmbed({ type, children }, base, contentext, resourceext) {
 }
 
 function embed(uri, node, whitelist = '', datawhitelist = '', logger) {
-  if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, datawhitelist.split(','))) {
+  if ((uri.scheme === 'http' || uri.scheme === 'https') && mm.some(uri.host, datawhitelist.split(',').map((s) => s.trim()))) {
```
Comment (tripodsan) [inline src/html/find-embeds.js:95]: ok, but I would convert the whilelists to an array in the `find` function, as it is needed several times.

**Pull request merged**